PORT-13795 - Enrich entities using AI (#2148)
port-labs/port-docs: docs/guides/all/enrich-entities-using-ai.md

---
displayed_sidebar: null
description: Learn how to enhance your entities with AI-powered insights from external APIs in Port
---
import ExistingSecretsCallout from '/docs/guides/templates/secrets/_existing_secrets_callout.mdx'

# Enrich entities using AI

> ## Overview

Reviewer: ```suggestion
```

> This guide will help you implement a self-service action in Port that uses AI to enrich entity properties with data from external APIs. This pattern enables you to automatically enhance your entities with intelligent insights from various AI services, making your software catalog more informative and actionable.

Reviewer: demonstrates how to...

> In this guide, we'll implement a security feature that uses OpenAI's API to analyze security issues and provide insights and possible remediation steps.

Reviewer: ```suggestion
In this guide, we will implement a security feature that uses OpenAI's API to analyze security issues and provide insights and possible remediation steps.
```

## Prerequisites

- Complete the [onboarding process](/getting-started/overview).

> - Access to OpenAI API with appropriate permissions (for this example).

Reviewer: what do you mean by "(for this example)"?

> - A blueprint for your entity type in Port (if you don't have one already).

Reviewer: ```suggestion
- A blueprint for your entity type in Port.
```

1. Go to your [Builder](https://app.getport.io/settings/data-model) page.
2. Search for the blueprint you want to update.
3. Click on the `{...}` button in the top right corner, and choose "Edit JSON".
4. Add this JSON snippet to the properties of the blueprint:

    ```json
          "ai_summary": {
            "type": "string",
            "title": "AI Analysis",
            "description": "AI-generated analysis and remediation steps",
            "format": "markdown"
          }
    ```
5. Click `Save`.

However, if you don't have a security integration installed, follow these steps to create a new blueprint with the structure below:

1. Go to your [Builder](https://app.getport.io/settings/data-model) page.
2. Click on the `+ Blueprint` button.
3. Click on the `{...}` button in the top right corner, and choose "Edit JSON".
4. Copy and paste the following JSON configuration into the editor.

>     ```json showLineNumbers

Reviewer: put this snippet in a collapsed box
